go/vt/vtgate/planbuilder/testdata/select_cases.txt

```diff
@@ -5891,40 +5841,19 @@ Gen4 plan same as above
   "QueryType": "SELECT",
   "Original": "SELECT music.id FROM music WHERE music.id IN (SELECT music.id FROM music WHERE music.foo = 'bar') AND music.user_id IN (3, 4, 5)",
   "Instructions": {
-    "OperatorType": "Subquery",
-    "Variant": "PulloutIn",
-    "PulloutVars": [
-      "__sq_has_values1",
-      "__sq1"
+    "OperatorType": "Route",
+    "Variant": "IN",
+    "Keyspace": {
```

Comment:
If I understand correctly, we should also be able to merge something like `SELECT music.id FROM music WHERE music.id IN (SELECT music.id FROM music WHERE music.foo = 'bar')` Could we add a test for this?

Reply:
I think we already have this test case here:

https://github.com/vitessio/vitess/blob/3719ffb79d56881049bc4ba014b4699d6280f655/go/vt/vtgate/planbuilder/testdata/select_cases.txt#L6111-L6170

That seems to be exactly the test you suggested? 🙇‍♂️

Reply:
Yep yep, that works 👍